naga/src/proc/constant_evaluator.rs

```diff
@@ -1193,6 +1193,8 @@ impl<'a> ConstantEvaluator<'a> {
                 UnaryOperator::Negate => match value {
                     Literal::I32(v) => Literal::I32(-v),
                     Literal::F32(v) => Literal::F32(-v),
+                    Literal::AbstractInt(v) => Literal::AbstractInt(-v),
+                    Literal::AbstractFloat(v) => Literal::AbstractFloat(-v),
```

Comment:
```suggestion
                    Literal::I32(v) => Literal::I32(v.wrapping_neg()),
                    Literal::F32(v) => Literal::F32(-v),
                    Literal::AbstractInt(v) => Literal::AbstractInt(v.wrapping_neg()),
                    Literal::AbstractFloat(v) => Literal::AbstractFloat(-v),
```

This is an existing issue but the spec says:

> If `T` is an integer scalar type and `e` evaluates to the largest negative value, then the result is `e`.

Reply:
Oh, good catch!